### PR TITLE
Azren/split up main

### PR DESCRIPTION
**Pull request by Azrenbeth**
Moved code from inside the main function into function calls

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -92,9 +92,19 @@ impl FromStr for LevelSizes {
     }
 }
 
-fn main() {
-    #[allow(deprecated)]
-    let matches = App::new(crate_name!())
+struct Config {
```
Comment (erikjohnston) [inline src/main.rs:95]: I know that the code base doesn't really have them atm, but can you add doc comments to the new structs and functions to say what they do please?

**Review by erikjohnston — COMMENTED** [review]
Other than needing some doc comments looks good!

File: src/main.rs (hunk 1 above)
Comment (Azrenbeth) [reply src/main.rs:95]: Ah sorry they're added in azren/documented_lib

Comment (erikjohnston) [reply src/main.rs:95]: Woop!

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**